includes/pixel.php

```diff
@@ -14,7 +14,7 @@ function wprtt_get_referring_page_title( $url, $post_id ) {
 	if ( ! is_wp_error( $response ) ) {
 
 		// find the title element inside of the response body.
-		$response = preg_match( '/<title.[^>]*>(.*)<\/title>/siU', $response['body'], $title_matches );
+		$response = preg_match( '/<title.[^>]*>([^<]*)<\/title>/siU', $response['body'], $title_matches );
```

Comment:
If I understand regex correctly (and that's debatable) I believe the regex to parse the title from the HTML should be more like this:

```suggestion
		$response = preg_match( '/<title[^>]*>(.*)<\/title>/iU', $response['body'], $title_matches );
```

I think the issue with the regex as written was that `title.` would expect there to be at least a character after `title`, so an opening title tag that's simply `<title>` and not `<title some-attribute>` won't match. Also, in [regex101](https://regex101.com/) the `/s` flag causes a "Catastrophic backtracking" error due to the extra dot in `title.` if there are no matches (e.g. if there's only `<title>` and not `<title some-attribute>`. The suggested change above does match both `<title>Article title</title>` and `<title some-attribute>Article title</title>` in my testing.

But to go a bit deeper, this still relies on regex first to fetch the article title, which seems expensive and unnecessary to me—it requires a `GET` request to fetch the whole page source. I'm not entirely sure if the regex to fetch the title from the page source is even necessary if we can fetch it by post ID instead (added in #183). The regex fetch pre-dates when we took ownership of the plugin, so I left it in place mostly untouched for compatibility. But perhaps we should update the regex as above and also flip the conditional so that we attempt to fetch the page title using `get_the_title( $post_id )` first, and fall back to the regex only if that fails for some reason?

Reply:
You're right! I mised that dot! And my regex was not fixing anything, it was just forcing the fallover to getting by ID.

I agree we could use the post ID there...

There is a difference that I don't know how relevant it is. But the idea here is that we are fetching the title of the republished post, and not the title from the original post...

But also, I'm seeing that in many cases the REFERRER is pointing to the site's home page, and not to the article page.. so we are actually getting the title of the site's homepage... so it's inconsistent anyways

Reply:
@benlk Thanks for clarifying and explaining the logic here! That makes a lot of sense and makes the test scenario much clearer. I think I misunderstood the purpose of this function in 7c58137, hence the fallback fetching a different piece of data. I think we should remove that fallback logic in addition to improving the regex to avoid the catastrophic backtracking.

Reply:
543bbd3 implements the fixed regex and removes the fallback to (incorrectly) get the shared article title if we can't get the referring URL's title.

Reply:
I can't approve my own PR but :+1:

Reply:
Wait, I did one last little tweak in https://github.com/Automattic/republication-tracker-tool/pull/228/commits/d57457915919a4e33389adf8a39caf0ba70f8116

Make sure it always return a string, especially now that we are calling `substr` for all params down there, we don't want to throw any warning or errors in case we can't fetch the title.